trino/serial.go

```diff
@@ -92,7 +107,19 @@ func Serial(v interface{}) (string, error) {
 	case []byte:
 		return "", UnsupportedArgError{"[]byte"}
 
+	case Date:
+		return "DATE " + time.Time(x).Format("'2006-01-02'"), nil
```

Comment:
You might create a `Date` somewhere from a `time.Time` that you created earlier or received from another part of a program and not know it has a non-zero time value. Consider a function like this:
```
func getOrders(since time.Time) []Order {
    rows, err := conn.Query(`SELECT * FROM orders WHERE created_at > ?`, trino.Date(since))
    // ...
}
```
This is error-prone, and the driver should not allow this. If we'd require the callers to explicitly create the Date from parts of `time.Time`, like `trino.Date(since.Year(), since.Month(), since.Day())` it's obvious to the caller that the time part is being ignored.

Reply:
Done